pkg/endpoint/bpf.go

```diff
@@ -1110,26 +1108,17 @@ func (e *Endpoint) deletePolicyKey(keyToDelete policy.Key, incremental bool, had
 		return false
 	}
 
-	var entry policy.MapStateEntry
-	var ok bool
-	if entry, ok = e.realizedPolicy.PolicyMapState[keyToDelete]; ok && entry.ProxyPort != 0 && hadProxy != nil {
-		*hadProxy = true
-	}
-
-	// Operation was successful, remove from realized state.
-	delete(e.realizedPolicy.PolicyMapState, keyToDelete)
 	e.updatePolicyMapPressureMetric()
 
 	e.PolicyDebug(logrus.Fields{
-		logfields.BPFMapKey:   keyToDelete,
-		logfields.BPFMapValue: entry,
-		"incremental":         incremental,
-	}, "deletePolicyKey")
+		logfields.BPFMapKey: keyToDelete,
+		"incremental":       incremental,
```

Comment:
The log messaging printing both key and entry with owners and dependents was useful during debugging. Since we are retrieving entry at the consumer site, can we add a new log there?